pkg/resource/deploy/source_eval.go

```diff
@@ -959,6 +961,20 @@ func (rm *resmon) RegisterResource(ctx context.Context,
 		t = tokens.Type(req.GetType())
 	}
 
+	// We handle update the provides map to include the providers field of the parent if
```

Comment:
Nit:

```suggestion
	// We handle updating the providers map to include the providers field of the parent if
```

? or was something else intended